scripts/foundationdb/init-foundationdb.sh

```diff
@@ -0,0 +1,238 @@
+        if [ -z "${1}" ]; then
+            usage
+        fi
+        HOST_TO_LISTEN_ON="${1}"
+        shift
+        continue
+    fi
+    if [ "${1}" = "--server-processes-num" ]; then
+        shift
+        if [ -z "${1}" ]; then
+            usage
+        fi
+        SERVER_PROCESSES_NUM="${1}"
+        shift
+        continue
+    fi
+    if [ "${1}" = "--data-dir" ]; then
+        shift
+        if [ -z "${1}" ]; then
+            usage
+        fi
+        DATA_DIR="${1}"
+        shift
+        continue
+    fi
+    if [ "${1}" = "--fdbcli-command" ]; then
+        shift
+        if [ -z "${1}" ]; then
+            usage
+        fi
+        FDBCLI_COMMAND="${1}"
+        shift
+        continue
+    fi
+    usage
+done
+
+if [ -z "${CLUSTER_FILE_CONTENT}" ] || [ -z "${HOST_TO_LISTEN_ON}" ]; then
+    usage
+fi
+
+
+########################
+### Helper functions ###
+########################
+
+APT_CACHE='/var/cache/apt/archives'
+PACKAGE_MIRROR='http://s3.amazonaws.com/appscale-build'
+
+log() {
+    local LEVEL=${2:-INFO}
+    echo "$(date +'%Y-%m-%d %T') $LEVEL $1"
+}
+
+cachepackage() {
+    cached_file="${APT_CACHE}/$1"
+    remote_file="${PACKAGE_MIRROR}/$1"
+    expected_md5="$2"
+    mkdir -p ${APT_CACHE}
+    if [ -f ${cached_file} ]; then
+        md5=($(md5sum ${cached_file}))
+        if [ "$md5" = "$2" ]; then
+            return 0
+        else
+            log "Incorrect md5sum for ${cached_file}. Removing it." "ERROR"
+            rm ${cached_file}
+        fi
+    fi
+
+    log "Fetching ${remote_file}"
+    if ! curl -fs "${remote_file}" > "${cached_file}"; then
+        log "Error while downloading ${remote_file}" "ERROR"
+        return 1
+    fi
+
+    actual_md5=($(md5sum ${cached_file}))
+    if [ "${actual_md5}" = "${expected_md5}" ]; then
+        return 0
+    else
+        log "md5 sum of downloaded file is ${actual_md5} though ${expected_md5} was expected" "ERROR"
+        log "Try downloading package manually to ${cached_file} and running script again"
+        rm ${cached_file}
+        return 1
+    fi
+}
+
+
+#####################################
+### Actual installation procedure ###
+#####################################
+
+SCRIPT_DIR="$( realpath --strip "$( dirname "${BASH_SOURCE[0]}" )" )"
+
+
+### Installing FDB clients package ###
+#------------------------------------#
+FDB_CLIENTS_PKG='foundationdb-clients_6.1.8-1_amd64.deb'
+FDB_CLIENTS_MD5='f701c23c144cdee2a2bf68647f0e108e'
+log "Making sure ${FDB_CLIENTS_PKG} is in ${APT_CACHE}"
+cachepackage "${FDB_CLIENTS_PKG}" "${FDB_CLIENTS_MD5}"
+
+log "Installing ${FDB_CLIENTS_PKG}"
+dpkg --install ${APT_CACHE}/foundationdb-clients_6.1.8-1_amd64.deb
+
+
+### Installing FDB server package ###
+#-----------------------------------#
+FDB_SERVER_PKG='foundationdb-server_6.1.8-1_amd64.deb'
+FDB_SERVER_MD5='80a427be14a329d864a91c9ce464d73c'
+log "Making sure ${FDB_SERVER_PKG} is in ${APT_CACHE}"
+cachepackage "${FDB_SERVER_PKG}" "${FDB_SERVER_MD5}"
```

Comment:
What do you think about installing the server package on our images so that someone trying appscale on vagrant, for example, could still start things without an internet connection?

Reply:
I like it. But I think we still need to leave package downloading here as the script could be used on any machine, not necessarily appscale vm. And using cache should help to avoid downloading if deb is already in place.

I'm thinking to move fdb initialization script to separate repo (https://github.com/AppScale/appscale-thirdparties), so I would add fdb packages downloading to appscale_install_functions.sh as a separate PR.

Reply:
The separate repo seems reasonable. In that case, I don't think we would need to install the server package when building the main "appscale" repo. The appscale-thirdparties repo could contain a build script that installs what it needs. I mainly want to avoid fetching and installing things when it comes time to run them since there's a lot that can go wrong during that process.